provider/pkg/gen/schema.go

```diff
@@ -1386,6 +1386,21 @@ func (g *packageGenerator) genTypeSpec(typeName, propName string, prop *discover
 			Type: "object",
 			Ref:  referencedTypeName,
 		}, nil
+	case prop.Type == "object" && prop.AdditionalProperties != nil:
+		// The prop is a map with a string key and a simple value
+		// if it doesn't have a ref.
+		if prop.AdditionalProperties.Ref == "" {
+			return &schema.TypeSpec{
+				Type: "object",
+				AdditionalProperties: &schema.TypeSpec{
+					Type: "string",
+				},
+			}, nil
+		}
+
+		// Otherwise, the value in-turn is a complex type.
+		typePropName := fmt.Sprintf(`%s%s`, typeName, strings.Title(propName))
+		return g.genTypeSpec(typePropName, propName, prop.AdditionalProperties, isOutput)
```

Comment:
I think this ends up just return the type of the value rather than a map from string -> valueType